src/mlpack/methods/ann/init_rules/glorot_init.hpp

```diff
@@ -98,28 +115,53 @@ inline void GlorotInitializationType<false>::Initialize(arma::Mat<eT>& W,
                                                        const size_t cols)
 {
   if (W.is_empty())
-  W = arma::mat(rows, cols);
+    W = arma::mat(rows, cols);
```

Comment:
I don't think it matters, but you use `set_size()` everywhere. In my opinion `arma::mat()` should work as well. So, it's up to you.
```suggestion
    W.set_size(rows, cols);
```